### ISSUE TITLE
Adding pip flags in environment.yml

### ISSUE THREAD

**Issue by tdhopper** [conda/conda#3763]
I have a dependency on a privately hosted PyPI server. I can pip install it by specifying the address with `--extra-index` when I do a pip install.

However, if I add pip flags under `pip:` in an environment.yml file, I get an error:

```
14:53 $ conda env update
Using Anaconda API: https://api.anaconda.org
Fetching package metadata ...........
Solving package specifications: ..........

Usage:
  pip install [options] <requirement specifier> [package-index-options] ...
  pip install [options] -r <requirements file> [package-index-options] ...
  pip install [options] [-e] <vcs project url> ...
  pip install [options] [-e] <local project path> ...
  pip install [options] <archive url/path> ...

no such option: --index-url http://HOST:8080/  --trusted-host HOST package


CondaValueError: Value error: pip returned an error.
```

Is there a good reason this has to fail? Can the parser just pass the pip flags along?

**Comment by mikeholler** [comment]
I'm pretty interested in specifying an alternate package index in conda `environment.yaml` files as well. Is the `pip:` clause in `environment.yaml` just passed directly as arguments to `pip`? I can find little documentation on this (specifying an alternate pypi repo doesn't seem like something a lot of people do) but it would be hugely useful to us if conda worked like this or provided some facility for doing such a thing.

**Comment by Juanlu001** [comment]
This would also be useful for installing Python packages directly from GitHub repositories, in case it's needed.

**Comment by fabboe** [comment]
That would be a nice and clean management to pass a private index for certain pip dependencies indeed.

A workaround similar to [this](http://stackoverflow.com/a/38187246/1836416) is to pass the full package URL on the private index.

**Comment by sdvillal** [comment]
All these concerns should be solved if something like #3969 is merged. When prompted by github bots, I'm still keeping that PR in a decent state for review/merge. But after a few months of not grabbing the admins attention, it is just being buried down in the PR list.

In the meantime, these workarounds (using full packages URLs, hideous syntax tricks like ```"--editable=git+https://github.com/neithere/argh.git#egg=argh"```, just using pip after the environment file...), although not pretty, will easily get the work done.

@kalefranz is there a chance to get some comments on #3969? It would be really appreciated.

**Comment by Casyfill** [comment]
how would I add `--upgrade` flag?
`"git+https://github.com/neithere/argh.git#egg=argh --upgrade"` does not work...
